packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1826,6 +1828,22 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
         _cursorTimer = null;
       }
     }
+
+    // Hide the text selection toolbar on mobile when orientation changes.
+    final Orientation orientation = MediaQuery.of(context).orientation;
+    if (_lastOrientation == null) {
+      _lastOrientation = orientation;
+      return;
+    }
+    if (orientation != _lastOrientation) {
+      _lastOrientation = orientation;
+      if (defaultTargetPlatform == TargetPlatform.iOS) {
```

Comment:
Nit: Maybe a `switch` on the platform here if you think that would be better?

Reply:
I considered it but this is more compact and I think it's just as readable.